docs/logging_snippets.py

```diff
@@ -0,0 +1,413 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Testable usage examples for Stackdriver Logging API wrapper
+
+Each example function takes a ``client`` argument (which must be an instance
+of :class:`google.cloud.pubsub.logging.Client`) and uses it to perform a task
+with the API.
+
+To facilitate running the examples as system tests, each example is also passed
+a ``to_delete`` list;  the function adds to the list any objects created which
+need to be deleted during teardown.
+"""
+
+import time
+
+from google.cloud.exceptions import NotFound
+from google.cloud.logging.client import Client
+
+
+def snippet(func):
+    """Mark ``func`` as a snippet example function."""
+    func._snippet = True
+    return func
+
+
+def _millis():
+    return time.time() * 1000
+
+
+def do_something_with(entry):  # pylint: disable=unused-argument
+    pass
+
+
+@snippet
+def logger_usage(client, to_delete):
+    """Exercise a logger instance."""
+    LOGGER_NAME = 'logger_usage-%d' % (_millis(),)
+
+    # [START logger_create]
+    logger = client.logger(LOGGER_NAME)
+    # [END logger_create]
+    to_delete.append(logger)  # in case of errors before explicit call
+
+    # [START logger_log_text]
+    logger.log_text("A simple entry")  # API request
+    # [END logger_log_text]
+
+    # [START logger_log_struct]
+    logger.log_struct({
+        'message': 'My second entry',
+        'weather': 'partly cloudy',
+    })  # API request
+    # [END logger_log_struct]
+
+    # [START logger_list_entries]
+    entries, token = logger.list_entries()  # API request
+    while True:
+        for entry in entries:
+            do_something_with(entry)
+        if token is None:
+            break
+        entries, token = logger.list_entries(page_token=token)  # API request
+    # [END logger_list_entries]
+
+    to_delete.remove(logger)
+    backoff = [1, 2, 4, 8]
```

Comment:
This is because our `retry` stuff is silo-ed? I'm still :+1: on making `system_tests` a local package (`setup.py` and all)